packages/@ember/-internals/routing/lib/services/router.ts

```diff
@@ -393,6 +394,32 @@ export default class RouterService extends Service {
     return this._router._routerMicrolib.recognizeAndLoad(internalURL);
   }
 
+  /**
+   * Refreshes all currently active routes, doing a full transition.
+   * If a routeName is provided and refers to a currently active route,
+   * it will refresh only that route and its descendents.
+   * Returns a promise that will be resolved once the refresh is complete.
+   * All resetController, beforeModel, model, afterModel, redirect, and setupController
+   * hooks will be called again. You will get new data from the model hook.
+   *
+   * @method refresh
+   * @param {String} [routeName] the route to refresh (along with all child routes)
+   * @return Transition
+   * @public
+   */
+  refresh(routeName?: string) {
```

Comment:
[The RFC named this `pivotRouteName`.](https://emberjs.github.io/rfcs/0631-refresh-method-for-router-service.html#detailed-design) which I found a little confusing since it's a new concept that isn't used anywhere else. It's named like that in router.js as well, but I'm not sure if it should be exposed like that to the end user.

Using `routeName` is just a personal preference and I'm not sure how much room for deviation there is when implementing an RFC, so let me know if I should change it back :smile:.

Reply:
Renamed it back to pivotRouteName anyway, since that is how it was accepted.